lib/src/main/java/io/cloudquery/internal/servers/plugin/v3/PluginServer.java

```diff
@@ -9,4 +25,100 @@ public class PluginServer extends PluginImplBase {
     public PluginServer(Plugin plugin) {
         this.plugin = plugin;
     }
+
+    @Override
+    public void getName(io.cloudquery.plugin.v3.GetName.Request request,
+            StreamObserver<io.cloudquery.plugin.v3.GetName.Response> responseObserver) {
+        responseObserver
+                .onNext(io.cloudquery.plugin.v3.GetName.Response.newBuilder().setName(plugin.getName()).build());
+        responseObserver.onCompleted();
+    }
+
+    @Override
+    public void getVersion(io.cloudquery.plugin.v3.GetVersion.Request request,
+            StreamObserver<io.cloudquery.plugin.v3.GetVersion.Response> responseObserver) {
+        responseObserver.onNext(
+                io.cloudquery.plugin.v3.GetVersion.Response.newBuilder().setVersion(plugin.getVersion()).build());
+        responseObserver.onCompleted();
+    }
+
+    @Override
+    public void init(io.cloudquery.plugin.v3.Init.Request request,
+            StreamObserver<io.cloudquery.plugin.v3.Init.Response> responseObserver) {
+        plugin.init();
+        responseObserver.onNext(io.cloudquery.plugin.v3.Init.Response.newBuilder().build());
+        responseObserver.onCompleted();
+    }
+
+    @Override
+    public void getTables(io.cloudquery.plugin.v3.GetTables.Request request,
+            StreamObserver<io.cloudquery.plugin.v3.GetTables.Response> responseObserver) {
+        try {
+            List<Table> tables = plugin.tables();
+            List<ByteString> byteStrings = new ArrayList<>();
+            for (Table table : tables) {
+                try (BufferAllocator bufferAllocator = new RootAllocator()) {
+                    Schema schema = table.toArrowSchema();
+                    VectorSchemaRoot schemaRoot = VectorSchemaRoot.create(schema, bufferAllocator);
+                    try (ByteArrayOutputStream out = new ByteArrayOutputStream()) {
+                        try (ArrowStreamWriter writer = new ArrowStreamWriter(schemaRoot, null,
+                                Channels.newChannel(out))) {
+                            writer.start();
+                            writer.end();
```

Comment:
I was expecting a call to a `write` method here, or is the call to `end` writing data?

Reply:
Good question, `start` writes the schema which is what we want here. We don't need to write any records (this will happen during `sync`) so there's not `write`.
https://github.com/apache/arrow/blob/6357c9f2419d5b0717e62adc8233c649e10de34b/java/vector/src/main/java/org/apache/arrow/vector/ipc/ArrowWriter.java#L183